domain/model/state/state.go

```diff
@@ -978,6 +978,145 @@ OR uuid IN (SELECT grant_on
 	return rval, nil
 }
 
+// ListModelSummariesForUser lists model summaries of all models the user has
+// access to. If no models are found then a nil slice is returned.
+// TODO(aflynn): 05-08-2024 - The ModelSummary struct includes a machine count,
+// unit count and cpu core count, model status as well as migration status. This
+// information has not yet been migrated over to the relational database. Once
+// it has, it needs to be included here.
+func (s *State) ListModelSummariesForUser(ctx context.Context, userName string) ([]model.ModelSummary, error) {
+	db, err := s.DB()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	q := `
+SELECT    (p.access_type, m.uuid, m.name, m.cloud_name, m.cloud_region_name, 
+          m.model_type, m.cloud_type, m.owner_name, m.cloud_credential_name, 
+          m.cloud_credential_cloud_name, m.cloud_credential_owner_name,
+          m.life, mll.time, m.target_agent_version) AS (&dbModelSummary.*)
+FROM      v_user_auth u
+JOIN      v_permission p ON p.grant_to = u.uuid
+JOIN      v_model m ON m.uuid = p.grant_on
+LEFT JOIN model_last_login mll ON m.uuid = mll.model_uuid
+WHERE     u.removed = false
+AND       u.disabled = false
```

Comment:
I don't think we need the disabled check here. `disabled` stops the user authenticating not the presence of data relation within the system. Disabled is a concern of authn not business logic.

Reply:
Good to know, will remove that